js/ui/grid_core/ui.grid_core.rows.js

```diff
@@ -830,7 +830,9 @@ module.exports = {
                                     const rowsHeight = that._getRowsHeight(contentElement.children().first());
                                     const $tableElement = $table || that.getTableElements();
                                     const borderTopWidth = Math.ceil(parseFloat($tableElement.css('borderTopWidth')));
-                                    const heightCorrection = browser.webkit && that._getDevicePixelRatio() >= 2 ? 1 : 0; // T606935
+                                    const isZoomedWebkit = browser.webkit && that._getDevicePixelRatio() >= 2; // T606935
+                                    const hasExtraBorderTop = browser.mozilla && browser.version >= 70 && !that.option('showRowLines');
+                                    const heightCorrection = isZoomedWebkit || hasExtraBorderTop ? 1 : 0;
```

Comment:
To my opinion it's is good practice to extract this corrections into a separate method (getHeightCorrection e.g.)

Reply:
Done